tests/chainer_tests/functions_tests/pooling_tests/test_average_pooling_2d.py

```diff
@@ -43,102 +32,33 @@ def _to_fcontiguous(arrays):
         {'use_chainerx': True, 'chainerx_device': 'native:0'},
         {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
     ])
-class TestAveragePooling2D(unittest.TestCase):
+class TestAveragePooling2D(testing.FunctionTestCase):
 
     def setUp(self):
-        dtype = self.dtype
-        x = numpy.random.uniform(-1, 1, (2, 3, 4, 3)).astype(dtype)
-        gy = numpy.random.uniform(-1, 1, (2, 3, 2, 2)).astype(dtype)
-        ggx = numpy.random.uniform(-1, 1, (2, 3, 4, 3)).astype(dtype)
-
-        self.output_shape = gy.shape
-
-        self.inputs = [x]
-        self.grad_outputs = [gy]
-        self.grad_grad_inputs = [ggx]
-
-        self.check_forward_options = {}
-        self.check_backward_options = {'dtype': numpy.float64}
+        self.input_shape = (2, 3, 4, 3)
+        self.output_shape = (2, 3, 2, 2)
```

Comment:
It looks like you can remove these two shapes from the class attribute since they're only used once, and "inline" them instead.

Reply:
Or better, how about parametrizing over common/uncommon shapes?